cpg_workflows/inputs.py

```diff
@@ -135,25 +135,32 @@ def _populate_analysis(cohort: Cohort) -> None:
             dataset=dataset.name,
         )
 
-        # NOTE: This logic will be simplified to remove existence checks overwriting metamist
-        # in a later PR.
         for sequencing_group in dataset.get_sequencing_groups():
             if (analysis := gvcf_by_sgid.get(sequencing_group.id)) and analysis.output:
-                assert analysis.output == sequencing_group.make_gvcf_path().path, (
+                # assert file exists
+                assert analysis.output.exists(), (
+                    'gvcf file does not exist',
                     analysis.output,
-                    sequencing_group.make_gvcf_path().path,
                 )
-                sequencing_group.gvcf = sequencing_group.make_gvcf_path()
+                sequencing_group.gvcf = analysis.output
             elif sequencing_group.make_gvcf_path().exists():
-                sequencing_group.gvcf = sequencing_group.make_gvcf_path()
+                logging.warning(
```

Comment:
So to clarify this behaviour, if metamist doesn't have an entry, and the gvcf path exists, we safely skip.

Reply:
Yes that's correct. Although, as Matt points down below, we might want to be stricter here and error out?